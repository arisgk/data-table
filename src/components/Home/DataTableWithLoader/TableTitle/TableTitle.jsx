import React from 'react';
import PropTypes from 'prop-types';
import IconButton from 'material-ui/IconButton';
import * as schemas from '../../../../schemas/propTypes';

const styles = {
  container: {
    height: 54,
    minHeight: 54,
    display: 'flex',
    alignItems: 'center',
  },
  title: {
    marginLeft: 24,
    marginTop: 0,
    marginBottom: 0,
    color: 'rgba(0, 0, 0, 0.54)',
    fontSize: '160%',
    fontWeight: 100,
  },
  iconButtonsContainer: {
    marginLeft: 'auto',
    marginRight: 24,
  },
  icon: {
    opacity: '0.54',
  },
};

const TableTitle = ({ selection, onStartDeleting }) => (
  <div style={styles.container}>
    <h2 style={styles.title}>Basketball Players</h2>
    <div style={styles.iconButtonsContainer}>
      <IconButton
        disabled={!(selection.selected && selection.selected.length > 0)}
        onTouchTap={() => onStartDeleting(selection.selected)}
        iconStyle={styles.icon}
      >
        <i
          className="material-icons"
          disabled={!(selection.selected && selection.selected.length > 0)}
        >
          delete
        </i>
      </IconButton>
    </div>
  </div>
);

TableTitle.propTypes = {
  selection: schemas.selection,
  onStartDeleting: PropTypes.func,
};

TableTitle.defaultProps = {
  selection: {},
  onStartDeleting: () => {},
};

export default TableTitle;
