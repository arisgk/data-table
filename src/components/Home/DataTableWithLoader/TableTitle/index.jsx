import TableTitle from './TableTitle';

export default TableTitle;
