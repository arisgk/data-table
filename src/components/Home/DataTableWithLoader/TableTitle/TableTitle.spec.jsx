/* global describe, it, expect */
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TableTitle from './TableTitle';

describe('DataTableWithLoader', () => {
  it('renders correctly', () => {
    const selection = { selected: [] };

    let wrapper = shallow(
      <TableTitle selection={selection} />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();

    selection.selected = ['1234abcd'];

    wrapper = shallow(
      <TableTitle selection={selection} />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();

    selection.selected = ['1234abcd', '5678efgh'];

    wrapper = shallow(
      <TableTitle selection={selection} />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();

    selection.selected = [];

    wrapper = shallow(
      <TableTitle selection={selection} />,
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
