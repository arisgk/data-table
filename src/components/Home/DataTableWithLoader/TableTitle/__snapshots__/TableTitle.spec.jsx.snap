// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTableWithLoader renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 54,
      "minHeight": 54,
    }
  }
>
  <h2
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.54)",
        "fontSize": "160%",
        "fontWeight": 100,
        "marginBottom": 0,
        "marginLeft": 24,
        "marginTop": 0,
      }
    }
  >
    Basketball Players
  </h2>
  <div
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": 24,
      }
    }
  >
    <IconButton
      disableTouchRipple={false}
      disabled={true}
      iconStyle={
        Object {
          "opacity": "0.54",
        }
      }
      onTouchTap={[Function]}
      tooltipPosition="bottom-center"
      touch={false}
    >
      <i
        className="material-icons"
        disabled={true}
      >
        delete
      </i>
    </IconButton>
  </div>
</div>
`;

exports[`DataTableWithLoader renders correctly 2`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 54,
      "minHeight": 54,
    }
  }
>
  <h2
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.54)",
        "fontSize": "160%",
        "fontWeight": 100,
        "marginBottom": 0,
        "marginLeft": 24,
        "marginTop": 0,
      }
    }
  >
    Basketball Players
  </h2>
  <div
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": 24,
      }
    }
  >
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={
        Object {
          "opacity": "0.54",
        }
      }
      onTouchTap={[Function]}
      tooltipPosition="bottom-center"
      touch={false}
    >
      <i
        className="material-icons"
        disabled={false}
      >
        delete
      </i>
    </IconButton>
  </div>
</div>
`;

exports[`DataTableWithLoader renders correctly 3`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 54,
      "minHeight": 54,
    }
  }
>
  <h2
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.54)",
        "fontSize": "160%",
        "fontWeight": 100,
        "marginBottom": 0,
        "marginLeft": 24,
        "marginTop": 0,
      }
    }
  >
    Basketball Players
  </h2>
  <div
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": 24,
      }
    }
  >
    <IconButton
      disableTouchRipple={false}
      disabled={false}
      iconStyle={
        Object {
          "opacity": "0.54",
        }
      }
      onTouchTap={[Function]}
      tooltipPosition="bottom-center"
      touch={false}
    >
      <i
        className="material-icons"
        disabled={false}
      >
        delete
      </i>
    </IconButton>
  </div>
</div>
`;

exports[`DataTableWithLoader renders correctly 4`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "height": 54,
      "minHeight": 54,
    }
  }
>
  <h2
    style={
      Object {
        "color": "rgba(0, 0, 0, 0.54)",
        "fontSize": "160%",
        "fontWeight": 100,
        "marginBottom": 0,
        "marginLeft": 24,
        "marginTop": 0,
      }
    }
  >
    Basketball Players
  </h2>
  <div
    style={
      Object {
        "marginLeft": "auto",
        "marginRight": 24,
      }
    }
  >
    <IconButton
      disableTouchRipple={false}
      disabled={true}
      iconStyle={
        Object {
          "opacity": "0.54",
        }
      }
      onTouchTap={[Function]}
      tooltipPosition="bottom-center"
      touch={false}
    >
      <i
        className="material-icons"
        disabled={true}
      >
        delete
      </i>
    </IconButton>
  </div>
</div>
`;
